File: src/main/java/random/telegramhomebot/model/HostTimeLog.java
package random.telegramhomebot.model;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.GenericGenerator;
import org.hibernate.annotations.Type;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import java.sql.Timestamp;
import java.util.UUID;

@Entity
@Table(name = "host_time_logs")
public class HostTimeLog {

	@Id
	@GeneratedValue(generator = "UUID")
	@GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
	@Type(type = "org.hibernate.type.UUIDCharType")
	@Column(length = 36, columnDefinition = "varchar(36)", updatable = false, nullable = false)
	private UUID id;
	@CreationTimestamp
	@Column(updatable = false)
	private Timestamp createdDate;
	@ManyToOne
	@JoinColumn(name = "host_id")
	private Host host;
	@Column
	HostState state;

	public HostTimeLog() {
	}

	public HostTimeLog(Host host, HostState state) {
		this.host = host;
		this.state = state;
	}

	public UUID getId() {
		return id;
	}

	public void setId(UUID id) {
		this.id = id;
	}

	public Timestamp getCreatedDate() {
		return createdDate;
	}

	public void setCreatedDate(Timestamp createdDate) {
		this.createdDate = createdDate;
	}

	public Host getHost() {
		return host;
	}

	public void setHost(Host host) {
		this.host = host;
	}

	public HostState getState() {
		return state;
	}

	public void setState(HostState state) {
		this.state = state;
	}
}

File: src/main/java/random/telegramhomebot/model/HostTimeLog.kt
package random.telegramhomebot.model

import com.fasterxml.jackson.annotation.JsonBackReference
import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.GenericGenerator
import org.hibernate.annotations.Type
import java.sql.Timestamp
import java.util.*
import javax.persistence.*

@Entity
@Table(name = "host_time_logs")
class HostTimeLog(
    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    @Type(type = "org.hibernate.type.UUIDCharType")
    @Column(length = 36, columnDefinition = "varchar(36)", updatable = false, nullable = false)
    var id: UUID? = null,
    @CreationTimestamp @Column(updatable = false)
    var createdDate: Timestamp? = null,
    @JoinColumn(name = "host_id") @ManyToOne @JsonBackReference
    var host: Host,
    var state: HostState
) {
    constructor(host: Host, state: HostState) : this(null, null, host, state)
}
